sound/soc/sof/intel/hda-dai.c

```diff
@@ -201,38 +193,27 @@ static int hda_link_dma_hw_params(struct snd_pcm_substream *substream,
 	if (!hlink)
 		return -EINVAL;
 
+	if (hext_stream->hstream.direction == SNDRV_PCM_STREAM_PLAYBACK)
+		snd_hdac_ext_bus_link_set_stream_id(hlink, stream_tag);
+
```

Comment:
you could mention that the order of configuration is changing:
now you configure the stream_id before stream_reset.

Is there a reason for the sequence change?

Reply:
the main reason is to club the use of hlink together so that this can be moved to the op later on.

Reply:
@ujfalusi @bardliao there are two independent parts here:

snd_hdac_ext_bus_link_set_stream_id(hlink, stream_tag); configure the multi-link mapping of streams to DMAs.

hdax_ext_stream_reset/setup program the DMA proper.

It's two different levels that can be done in any order. The only required order is reset before setup.